invenio_app_ils/circulation/indexer.py

```diff
@@ -37,7 +38,14 @@ def index_referenced_records(loan):
         item = resolve_item_from_loan(loan["item_pid"])
         referenced.append(dict(pid_type=loan["item_pid"]["type"], record=item))
 
-    # index the document
+    # index the loan itself
```

Comment:
IMHO this comment is not enough, in one month I will be probably asking myself why we are doing this :)
```suggestion
    # index the loan itself: this is needed because of the extra field `available_items_for_loan_count`
    # added when indexing. To calculate the value of this field, a search on the `loans`
    # indexed is performed and this loan has to be already indexed with its latest data.
    # At the first indexing, `available_items_for_loan_count` value might be wrong and corrected
    # at the second re-indexing. 
```